mrcc/default.nix

```diff
@@ -0,0 +1,187 @@
+      sed -i "s!@out@/bin/!$(pwd)/${localRelPrefix}/!g" $exe
+    done;
+
+    # Set a umask to make sure all files created during MRCC run have correct permissions.
+    originalUmask=$(umask)
+    umask 0023
+
+    # The Basis directory must be copied and not just symlinked. This ensures, that the basis files
+    # have correct permissions in there.
+    cp --no-preserve=all -r @out@/share/BASIS ${localRelPrefix}/.
+
+    # Save the original old path variable to restore it later and then make sure the local MRCC
+    # installation prefix is the place where all the mrcc exes are found.
+    originalPATH=$PATH
+    export PATH=$(pwd)/${localRelPrefix}:$PATH
+
+    # Create a trap if MRCC fails.
+    trap cleanup EXIT ERR
+  '';
+  dmrccEntryScript = writeScript "dmrcc" (commonEntry + ''
+      $(pwd)/${localRelPrefix}/dmrcc-original
+    '');
+  dmrccMpiEntryScript = writeScript "dmrcc_mpi" (commonEntry + ''
+      $(pwd)/${localRelPrefix}/dmrcc_mpi-original
+    '');
+
+in
+  /*
+  This is the normal MRCC derivation without shenanigans. Only in the postFixup wrapper magic
+  happens to make MRCC work for the user. See the comments at postFixup.
+  */
+  stdenv.mkDerivation rec {
+    pname = "mrcc";
+    version = "2020.02.22";
+
+    nativeBuildInputs = [
+      makeWrapper
+    ];
+
+    propagatedBuildInputs = [
+      bash
+      intelmpi
+    ];
+
+    src = requireFile rec {
+      name = "mrcc.2020-02-22.binary.tar.gz";
+      sha256 = "e7b4944c66b1b127f9e7b47ca973eee189a6085d4bbc8b2e46b082e57e49ad0b";
+      url = "https://www.mrcc.hu/index.php/download-mrcc/mrcc-binary/send/4-mrcc-binary/31-mrcc-2020-02-22-binary-tar";
+      message = ''
+        The MRCC source code and binaries are not publically available. Obtain your own license at
+        https://www.mrcc.hu and download the binaries at ${url}. Add the archive ${name} to the nix
+        store by:
+          nix-store --add-fixed sha256 ${name}
+        and then rebuild.
+      '';
+    };
+
+    unpackPhase = ''
+      tar -xf $src
+    '';
+
+    dontConfigure = true;
+    dontBuild = true;
+    installPhase = ''
+      runHook preInstall
+
+
+      # Move executables to $out
+      mkdir -p $out/bin
+      exes=$(find -type f -executable)
+      for exe in $exes; do
+        cp $exe $out/bin/.
+      done
+
+      # Move the tests and basis sets to share and make a symlink for MRCC
+      mkdir -p $out/share
+      cp -r BASIS $out/share/.
+      cp -r MTEST $out/share/.
+      ln -s $out/share/BASIS $out/bin/.
+      ln -s $out/share/MTEST $out/bin/.
+
+      # Copy the manual also to share
+      cp manual.pdf $out/share/.
+
+
+      runHook postInstall
+    '';
+
+    # Manual fix of the RPATHs of the partially dynamically linked MPI programs
+    preFixup = ''
+      cd $out/bin
+      dynamicExes="dmrcc_mpi mrcc_mpi ccsd_mpi scf_mpi"
+      for exe in $dynamicExes; do
+        patchelf --set-rpath ${intelmpi}/lib:${stdenv.cc.cc.lib}/lib $exe
```

Comment:
the `autoPatchElfHook` fails here?

Reply:
yes, autoPatchelf fails on some of the static executables and produces a non-0 exit code.